### DISCUSSION TITLE
Create vite server with 'test' mode

### DISCUSSION THREAD

**Discussion by GalassoLuca** [vitest-dev/vitest#642] [Ideas]
I recently migrated more than 5 hundred tests from Jest to Vitest and I'm surprised about Vitests speed and reliability!

While running these tests I need to load env variables and I'm properly doing it by adding them in the `.env` file with the `VITE_` prefix as suggested from the doc.

In order to clarify which vars are used while testing and which ones are not, I'd like to split them into two different files. This is possible by using [VITE mode](https://vitejs.dev/guide/env-and-mode.html#modes) while creating the Vite server with `createServer` Vite's method used in `createVitest` Vitest's method.

Currently, in Vitest the mode is not specified and so the default is `development`.
Knowing that I can split these variables by creating a `.env.development` file and a `.env` file.

My concern is: isn't better to specify the mode? Should it be set to `test` or specified in the `viteOverrides` param of `createVitest` function?

**Comment by 1pm** [comment]
I currently use `process.env.NODE_ENV` or `process.env.TEST` (released in 0.3.4) to check if running tests.
My tests still run vite in `development` mode and I haven't figured a way to change it.

There is https://github.com/vitest-dev/vitest/pull/713 released in 0.3.0 which should set mode to `test` if no value is set.
Is this possible to configure now? I tried various combinations like `vite run --mode=test` etc. with no success

**Reply by sheremet-va** [reply]
By `mode` what exactly do you mean? `mode` variable inside `defineConfig`?

**Reply by 1pm** [reply]
Yes, vite's [mode](https://vitejs.dev/config/#mode) so different configuration can be used for tests

**Reply by sheremet-va** [reply]
Right now we override it after config is defined. Will make a fix to pass mode to `defineConfig` function

**Reply by 1pm** [reply]
Great, thank you for your quick response

**Comment by GalassoLuca** [comment]
I was opening a new PR while I saw @sheremet-va's commits of the #775, published in [v0.4.0](https://github.com/vitest-dev/vitest/releases/tag/v0.4.0)

I tested it and it's working as expected!

**Reply by awacode21** [reply]
i think i may have a related bug: https://github.com/vitest-dev/vitest/issues/886

when i try to set --mode as command option it results in true (instead of the given value) in the defineConfig function
